lib/ace/ext/papeeria/highlighter.coffee

```diff
@@ -10,23 +10,28 @@ define((require, exports, module) ->
             session.removeMarker(session.$bracketHighlightRight)
             session.$bracketHighlightLeft = null
             session.$bracketHighlightRight = null
+            session.$highlightMode = false
+            return
         if !pos.mismatch
             rangeLeft = new Range(pos.left.row, pos.left.column, pos.left.row, pos.left.column + 1)
             rangeRight = new Range(pos.right.row, pos.right.column, pos.right.row, pos.right.column + 1)
             session.$bracketHighlightLeft = session.addMarker(rangeLeft, "ace_bracket", "text")
             session.$bracketHighlightRight = session.addMarker(rangeRight, "ace_bracket", "text")
+            session.$highlightMode = true
         else
             if pos.left && pos.right 
                 range = new Range(pos.left.row, pos.left.column, pos.right.row, pos.right.column + 1)
                 session.$bracketHighlightLeft = session.addMarker(range, "ace_error-marker", "text")
-
+                session.$highlightMode = true
```

Comment:
isn't it enough that we have $bracketHighlightLeft or right as indicator that we highlight something? Why do we need yet another flag?